yarn-project/aztec-node/src/aztec-node/aztec-node.ts

```diff
@@ -188,4 +183,24 @@ export class AztecNode {
     const leafIndex = computePublicDataTreeLeafIndex(contract, new Fr(slot), await PrimitivesWasm.get());
     return this.merkleTreeDB.getLeafValue(MerkleTreeId.PUBLIC_DATA_TREE, leafIndex, false);
   }
+
+  /**
+   * Returns the current committed roots for the data trees.
+   * @returns the current committed roots for the data trees.
+   */
+  public async getTreeRoots(): Promise<Record<MerkleTreeId, Fr>> {
```

Comment:
This information is actually already available within the AztecRPCServer as part of the L2Block download. I know we are building the Aztec Sandbox but there will come a point where we want to rely on Aztec Node for data as little as possible, so I wonder if it is worth removing this interaction in favour of taking the locally available version?

Reply:
Ah, I didn't know that! It makes a lot more sense. I'll add methods to the aztec-rpc `Synchroniser` to save in memory the latest roots and use those!